File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    var answer = [[]];
    for(let i=0; i<arr.length; i++){
        for(let j=0; j<arr[i].length; j++){
            if(arr.length > arr[i].length){
                while(arr.length !== arr[i].length){
                    arr[i].push(0);
                }
            }else{
                while(arr.length !== arr[i].length){
                    arr.push(new Array(arr.length).fill(0));
                }
            }
        }
    }
    return arr;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    row = 0
    for i in range(len(arr)):
        row = max(row,len(arr[i]))
    size = max(len(arr),row)
    
    if len(arr) < size:
        for _ in range(size - len(arr)): 
            arr.append([0] * size)
    
    for i in range(len(arr)):
        if len(arr[i]) < size:
            arr[i]+= [0]*(size-len(arr[i]))
    return arr